hive_metastore_client/hive_metastore_client.py

```diff
@@ -130,6 +136,21 @@ def add_partitions_to_table(
 
         self.add_partitions(partition_list_with_correct_location)
 
+    def create_database_if_not_exists(self, database: Database) -> None:
+        """
+        Creates the table in Hive Metastore if it does not exist.
+
+        Since hive metastore server and thrift mapping do not have the option
+         of checking if the database does not exist, this method simulates this
+         this behavior.
```

Comment:
```suggestion
It encapsulates the logic of checking if the database does not exist, so it can be created, since hive does not have a this built in command.
```
What if we rewrite it to be closer to what @jufreire wrote in the previous method?

Reply:
My suggestion were a bit off since the UI in git is trolling me haha. Just need to better indent it

Reply:
I thought about keeping the same style, but this method doesn't literally encapsulate the logic of checking. Instead of _encapsulating_ this behavior, the method _simulates_ this behavior.